File: problemset/Unique_Paths.cpp
class Solution {
public:
    vector<vector<int>> dp;
    int uniquePaths(int m, int n) {
        dp.resize(n, vector<int>(m, 0));
        for (int i = 0; i < m; ++i) dp[n-1][i] = 1;
        for (int i = n-2; i >= 0; --i)
            for (int j = m-1; j >= 0; --j) {
                dp[i][j] = dp[i+1][j];
                if (j != m-1) dp[i][j] += dp[i][j+1];
            }
        return dp[0][0];
    }
};

File: activity/20200904/scau-flyingbird/UniquePaths.go
func uniquePaths(m int, n int) int {
	var dp [2][100]int
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if i == 0 || j == 0 {
				dp[i%2][j] = 1
			} else {
				dp[i%2][j] = dp[i%2][j-1] + dp[(i-1)%2][j]
			}
		}
	}
	return dp[(m-1)%2][n-1]
}
